src/lib/support/PersistedCounter.h

```diff
@@ -124,20 +134,32 @@ class PersistedCounter : public MonotonicallyIncreasingCounter<T>
 
         ReturnErrorOnFailure(MonotonicallyIncreasingCounter<T>::Advance());
 
+        return VerifyAndPersistNextEpochStart(mNextEpoch);
+    }
+
+private:
+    CHIP_ERROR VerifyAndPersistNextEpochStart(T refEpoch)
+    {
+
         if (MonotonicallyIncreasingCounter<T>::GetValue() >= mNextEpoch)
```

Comment:
So fixing this check (which does not seem to be right in the "counter rolled over when doing AdvanceBy" case) is a followup, right?

Reply:
Fixed the check and added an extra unit test for `AdvanceBy` to validate the behavior.